source/adapters/native_cpu/command_buffer.cpp

```diff
@@ -0,0 +1,131 @@
+//===--------- command_buffer.cpp - NativeCPU Adapter ---------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+
+#include "common.hpp"
+
+/// Stub implementations of UR experimental feature command-buffers
+/// Taken almost unchanged from another adapter. Perhaps going forward
+/// these stubs could be defined in core UR as the default which would
+/// reduce code duplication. Adapters could then "override" these defaults.
```

Comment:
I'd prefer it if this design suggestion was turned into a GitHub Issue on the repo where it could be discussed with maintainers, rather than living as a comment.

Reply:
Thanks Ewan I've raised it as an issue here https://github.com/oneapi-src/unified-runtime/issues/1064